contents/textbook/lecture02/factoryMethods/factoryMethods.md

```diff
@@ -36,5 +36,13 @@ According to my understanding, factory method is defined as static so that it ca
 1. Constructors cannot be named, while readable names can be assigned to factory methods.
 2. Constructors can only return same type as the class name, however, factory methods can return other objects
 3. Factory methods promote the idea of using interface instead of implementation, and would generate more flexible codes.
-
+4. Unlike constructors, factory methods doesn't require a new object to be created everytime they are called.
+   - Instantiate a static object as a default value, say an empty box
+   - `private static final BOX EMPTY_BOX = new BOX();`
+   - Factory method just has to return this specific instance of empty box whenever required instead of creating tons of new empty boxes
```

Comment:
Perhaps you meant static method instead of factory method. Factory method is a method of constructing objects, rather than a method returning static initialised objects

Reply:
Hmm but from the question in lab 4 and [this stackoverflow](https://stackoverflow.com/questions/53579911/why-no-public-constructor-for-optional-in-java), i thought that Box.empty() would be a special type of factory method, only to serve the purpose of returning the special type of Box with an empty value (according to the question in lab4).

Reply:
Fair enough, you could put it that way

Reply:
That's interesting. Never thought of the empty method being considered a factory.

Reply:
Ikr